scan/lib/scan/test_command_generator.rb

```diff
@@ -31,8 +31,9 @@ def options
 
         options = []
         options += project_path_array
-        options << "-sdk '#{config[:sdk]}'" if config[:sdk]
-        options << destination # generated in `detect_values`
+        options << "-sdk '#{config[:sdk]}'" if config[:sdk] && !config[:build_for_testing]
+        options << destination unless config[:build_for_testing] # generated in `detect_values`
+        options << "  -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO" if config[:build_for_testing]
```

Comment:
if we build for testing we need `iphonesimulator` as a sdk - in the `config[:sdk]` user can supply one, for testing, but as we want to build only once, its required to have `iphonesimulator` as sdk.